### ISSUE TITLE
`setZone: true` doesn't work if timezone offset has minutes component

### ISSUE THREAD

**Issue by amCap1712** [moment/luxon#1365]
**Describe the bug**
`setZone: true` doesn't work if timezone offset has minutes component.

**To Reproduce**

```ts
import { DateTime } from "luxon";

let date1 = DateTime.fromISO("2023-01-19T13:00:00+06:30", {setZone: true});
let date2 = DateTime.fromISO("2023-01-19T13:00:00+06:00", {setZone: true});
console.log(date1.toLocaleString(DateTime.DATETIME_HUGE));
console.log(date2.toLocaleString(DateTime.DATETIME_HUGE));
```

**Actual vs Expected behavior**
Actual Output
```
Thursday, 19 January, 2023 at 6:30 am Coordinated Universal Time 
Thursday, 19 January, 2023 at 1:00 pm GMT+06:00 
```
Expected Output
```
Thursday, 19 January, 2023 at 1:00 pm GMT+06:30
Thursday, 19 January, 2023 at 1:00 pm GMT+06:00
```

**Desktop (please complete the following information):**

- OS: Ubuntu 22.10
- Browser: Firefox 108.0.2 
- Luxon version: 3.2.1
- Your timezone: Asia/Kolkata

**Comment by icambron** [comment]
The bug here is not that `setZone` isn't working, but rather that `toLocaleString` is not handling the zone. Also worth nothing that some time formats work (anything that doesn't include the zone in the output):

```js
let date1 = DateTime.fromISO("2023-01-19T13:00:00+06:30", {setZone: true}); 
date1.toISO()  //=> '2023-01-19T13:00:00.000+06:30'
date1.toLocaleString(DateTime.DATETIME_SHORT)   //=> '1/19/2023, 1:00 PM'                                                                                                                                     
date1.toLocaleString(DateTime.DATETIME_HUGE)   //=> 'Thursday, January 19, 2023 at 6:30 AM Coordinated Universal Time' 
```

This seems to be a limitation of Intl.DateTimeFormat, which doesn't support half-hour offsets:

```js
> new Intl.DateTimeFormat("en", { timeZone: "Etc/GMT-06:30" })                                                                                                     
Uncaught RangeError: Invalid time zone specified: Etc/GMT-06:30                                                                                                    
    at new DateTimeFormat (<anonymous>)  
```

This is the current logic for handling this:

```js
        // Not all fixed-offset zones like Etc/GMT+4:30 are present in tzdata.
        // So we have to make do. Two cases:
        // 1. The format options tell us to show the zone. We can't do that, so the best
        // we can do is format the date in UTC.
        // 2. The format options don't tell us to show the zone. Then we can adjust them
        // the time and tell the formatter to show it to us in UTC, so that the time is right
        // and the bad zone doesn't show up.
```

We are in case 1. I'm curious if we can do better than this though. It's tricky because "GMT-06:30" is not the same in every language. However, it might work to detect this and:

1. format to parts
2. replace the bad zone with GMT[+-]xxxx
3. write back out the string

...on the logic that an English string is better than using a different zone.

**Comment by diesieben07** [comment]
This was fixed by #1377, which was released with version 3.3.0.
```js
let date1 = DateTime.fromISO("2023-01-19T13:00:00+06:30", {setZone: true});
console.log(date1.toLocaleString(DateTime.DATETIME_HUGE));
```
This code now outputs `Thursday, January 19, 2023 at 1:00 PM UTC+6:30`.
